### PR TITLE
go: Replace x/maps package by respective standard libraries

### PR DESCRIPTION
**Pull request by sayboras**
Please refer to individual commit for more details.

### CODE REVIEW

**Review by gandro — APPROVED** [review]
Thank you!

File: pkg/fqdn/cache.go
Comment (doniacld) [inline pkg/fqdn/cache.go]: I would have kept `slices` for the standard package and renamed the `cilium` one but maybe it is not part of this PR 😇

**Review by doniacld — APPROVED** [review]
One minor comment otherwise LGTM!

File: pkg/endpointmanager/policymap_pressure.go
Comment (squeed) [inline pkg/endpointmanager/policymap_pressure.go:76]: Had to double-check that this was correct, since the types work out the same. But yes, this is good :-).

**Review by giorio94 — APPROVED** [review]
/lgtm for my codeowners, thanks!

**Review by michi-covalent — APPROVED** [review]
🥰

**Comment by sayboras** [comment]
Waiting for other reviews before resolving conflict on go.mod, just to avoid unnecessary CI re-run.

**Review by youngnick — APPROVED** [review]
LGTM from servicemesh POV.

File: pkg/fqdn/cache.go
Comment (sayboras) [reply pkg/fqdn/cache.go]: thanks, I need to resolve go.mod conflict, so will incorporate your comment as well.

File: pkg/hive/feature_lifecycle.go
Comment (joamaki) [inline pkg/hive/feature_lifecycle.go]: Is the `append` really needed? Only difference I can see is that this makes sure a `nil` slice isn't returned, but that's totally fine. I'd fix the tests if that's what's breaking.

File: pkg/maps/ctmap/ctmap_privileged_test.go
Comment (joamaki) [inline pkg/maps/ctmap/ctmap_privileged_test.go]: How about:
```
toDelete := size/4
for k := range cache {
   ...
   toDelete--
   if toDelete <= 0 { break } 
}
```

**Review by joamaki — APPROVED** [review]

File: pkg/hive/feature_lifecycle.go
Comment (sayboras) [reply pkg/hive/feature_lifecycle.go]: Yes, its mainly to avoid breaking current tests. I will remove this and fix the tests accordingly. Thanks

File: pkg/allocator/allocator_test.go
Comment (giorio94) [inline pkg/allocator/allocator_test.go]: Given that you are going to apply a few other suggestions as well, this could be simplified to:

```suggestion
	ids := slices.Sorted(maps.Keys(d.identities))
```

Comment (sayboras) [reply pkg/allocator/allocator_test.go]: Ah thanks, i thought i changed this part before 🫣

**Review by tklauser — APPROVED** [review]

**Review by pippolo84 — APPROVED** [review]
Thanks! :rocket:

**Review by harsimran-pabla — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Pull request merged**
